mayastor/src/grpc.rs

```diff
@@ -61,245 +83,324 @@ impl Mayastor for MayastorGrpc {
         &self,
         request: Request<CreatePoolRequest>,
     ) -> Result<Response<Null>> {
-        locally! { pool::create_pool(request.into_inner()) };
+        let args = request.into_inner();
+        trace!("{:?}", args);
+        let name = args.name.clone();
+
+        if args.disks.is_empty() {
+            return Err(Status::invalid_argument("Missing devices"));
+        }
+        debug!(
+            "Creating pool {} on {} with block size {}, io_if {}...",
+            name,
+            args.disks.join(" "),
+            args.block_size,
+            args.io_if,
+        );
+
+        locally! { pool::create_pool(args) };
+
+        info!("Created or imported pool {}", name);
         Ok(Response::new(Null {}))
     }
 
     async fn destroy_pool(
         &self,
         request: Request<DestroyPoolRequest>,
     ) -> Result<Response<Null>> {
-        locally! { pool::destroy_pool(request.into_inner()) };
+        let args = request.into_inner();
+        trace!("{:?}", args);
+        let name = args.name.clone();
+        debug!("Destroying pool {} ...", name);
+        locally! { pool::destroy_pool(args) };
+        info!("Destroyed pool {}", name);
         Ok(Response::new(Null {}))
     }
 
     async fn list_pools(
         &self,
-        _request: Request<Null>,
+        request: Request<Null>,
     ) -> Result<Response<ListPoolsReply>> {
+        let args = request.into_inner();
+        trace!("{:?}", args);
         assert_eq!(Cores::current(), Cores::first());
 
-        let pools = pool::list_pools()
-            .iter()
-            .map(|pool| Pool {
-                name: pool.name.clone(),
-                disks: pool.disks.clone(),
-                state: match pool.state.as_str() {
-                    "online" => PoolState::PoolOnline,
-                    _ => PoolState::PoolFaulted,
-                } as i32,
-                capacity: pool.capacity,
-                used: pool.used,
-            })
-            .collect();
-
-        Ok(Response::new(ListPoolsReply {
-            pools,
-        }))
+        let reply = ListPoolsReply {
+            pools: pool::list_pools()
+                .iter()
+                .map(|pool| Pool {
+                    name: pool.name.clone(),
+                    disks: pool.disks.clone(),
+                    state: match pool.state.as_str() {
```

Comment:
The whole idea of getting away from jsonrpc for our own methods is to get rid of these mappings, to make sure we do that I've filed an issue for this, thanks for the itermediate solution :+1: